File: CSharpCodeBase/utils/ailib/aitree.lua
local Class = require 'utils.hump.class'

local AITree = Class({init=function (self, entity, root)
  self.entity = entity
  self.root = root
end})

function AITree:Process()
  self.root:Visit()
  self.root:Save()
  self.root:Process()
end

function AITree:Reset()
  self.root:Reset()
end

function AITree:OnEvent(data)
  self.root:OnEvent(data)
end

return AITree

File: CSharpCodeBase/utils/ailib/aitree.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class aitree {
 public void init (self, entity, root);
   this.entity = entity;
   this.root = root;
 }})
 public void Process(){
   this.root:Visit();
   this.root:Save();
   this.root:Process();
 }
 public void Reset(){
   this.root:Reset();
 }
 public void OnEvent(data){
   this.root:OnEvent(data);
 }
 }}